capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -0,0 +1,256 @@
+import React, { Component } from 'react';
+
+import { Button, Form, Spinner, Modal, Col, Row, Card } from 'react-bootstrap'
+
+import '../styles/SearchUserModal.css'
+import '../styles/Modal.css'
+
+export class SearchUserModal extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      fetchingUsers: false, // For Spinner
+      showModal: false,
+      typingTimeout: 0,
+      searchTerm: '',
+      searchResults: [],
+      addedUsersTracker: [],
+      addedUsers: [],
+      addedFriends: [],
+      resultsFound: false,
+    }
+  }
+
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingUsers: true })
+
+    let searchResults;
+
+    if (searchTerm === '') {
+      searchResults = [];
+
+      this.setState({ searchResults, fetchingUsers: false, resultsFound: false })
+    }
+    else {
+      searchResults = await fetch(`/api/userSearch?searchTerm=${searchTerm}`)
+        .then(response => response.json())
+        .catch(err => alert(err));
+
+      if (typeof searchResults === 'undefined') {
+        searchResults = [];
+      }
+
+      if (searchResults.length !== 0) {
+        this.setState({ searchResults, fetchingUsers: false, resultsFound: true }) // Guilty
+      }
+      else {
+        this.setState({ searchResults, fetchingUsers: false, resultsFound: false }) // Not Guilty
+      }
+    }
+  }
+
+  handleSearchTermChange = (event) => {
+
+    // resultsFound innocent until proven Guilty
+    this.setState({ searchResults: [], resultsFound: true })
+
+    if (this.state.typingTimeout) {
+      clearTimeout(this.state.typingTimeout);
+    }
+
+    this.setState({
+      searchTerm: event.target.value,
+      typingTimeout: setTimeout(async () => {
+        await this.getBooks(this.state.searchTerm)
+      }, 500)
+    })
+  }
+
+  addUserToAddedUsers = (user) => {
+
+    if (!this.state.addedUsersTracker.includes(user)) {
+      this.setState({ addedUsersTracker: [...this.state.addedUsersTracker, user] })
+    }
+
+    // Rerender
+    this.setState({ addedUsers: [...this.state.addedUsers, user] })
+  }
+
+  addUserToAddedFriends = (user) => {
+    if (!this.state.addedUsersTracker.includes(user)) {
+      this.setState({ addedUsersTracker: [...this.state.addedUsersTracker, user] })
+    }
+
+    // Rerender
+    this.setState({ addedFriends: [...this.state.addedFriends, user] })
+  }
+
+  removeUserFromAddedUsers = (user) => {
+
+    const indexAddedUsers = this.state.addedUsers.indexOf(user);
+    this.state.addedUsers.splice(indexAddedUsers, 1)
```

Comment:
It might be better to make a deep copy of this array to avoid changing the state without set state at the end? I know that you have a setState at the end of the function but it's generally bad practice to change a state variable like this

Reply:
Will try to change in the next PR.

Reply:
You should be able to use a `filter` here to create a new array and then set the state to it